lib/internal/errors.js

```diff
@@ -35,21 +35,26 @@ class AssertionError extends Error {
     if (typeof options !== 'object' || options === null) {
       throw new exports.TypeError('ERR_INVALID_ARG_TYPE', 'options', 'object');
     }
-    if (options.message) {
-      super(options.message);
+    var { actual, expected, message, operator, stackStartFunction } = options;
+    if (message) {
+      super(message);
     } else {
+      if (actual && actual.stack && actual instanceof Error)
```

Comment:
Isn't `actual instanceof Error` itself enough?

Reply:
It is but it has a performance implication (it might not be important because it is something that I always like to consider).

Reply:
Hmmm.. actually, I believe `.stack` is a getter isn't it? It will have a performance impact also.

Reply:
@jasnell util.inspect will look at the stack as well. The first call to the call is heavy and it does not matter if it is done here or later on.

Reply:
So yes, it would be faster for Error objects (the instanceof check is added on top of the stack access) but slower for the average case.

Reply:
Yeah. Hmm. It's likely fine. Doing it the same way internal/util isError does would likely be ok also but I'm good with it :)